napari/layers/base/base.py

```diff
@@ -638,6 +639,13 @@ def block_update_properties(self):
         yield
         self._update_properties = True
 
+    @contextmanager
```

Comment:
I put this here because I thought we might want to use it on other layers. However, I didn't implement it on the other layers. In hindsight, maybe it's better to put it just on the Shapes layer for now and then if it becomes a more general strategy, we can move it back to the Base layer. Thoughts?